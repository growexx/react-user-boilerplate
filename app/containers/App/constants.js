/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

import {
  FAV_ICON_152_152,
  FAV_ICON_16_16,
  FAV_ICON_32_32,
  APPLE_TOUCH_ICON,
  FAV_ICON_192_192,
} from '../../images/favicons';
export const LOAD_REPOS = 'boilerplate/App/LOAD_REPOS';
export const LOAD_REPOS_SUCCESS = 'boilerplate/App/LOAD_REPOS_SUCCESS';
export const LOAD_REPOS_ERROR = 'boilerplate/App/LOAD_REPOS_ERROR';
export const LOAD_APP = 'boilerplate/App/LOAD_APP';
export const FAV_ICONS = [
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '512x512',
    href: FAV_ICON_152_152,
  },

  {
    rel: 'icon',
    type: 'image/png',
    sizes: '192x192',
    href: FAV_ICON_192_192,
  },
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '32x32',
    href: FAV_ICON_32_32,
  },
  {
    rel: 'icon',
    type: 'image/png',
    sizes: '16x16',
    href: FAV_ICON_16_16,
  },
  {
    rel: 'apple-touch-icon',
    href: APPLE_TOUCH_ICON,
  },
];
