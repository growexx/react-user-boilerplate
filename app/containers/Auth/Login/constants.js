export const CHANGE_EMAIL = 'boilerplate/Login/CHANGE_EMAIL';
export const CHANGE_PASSWORD = 'boilerplate/Login/CHANGE_PASSWORD';
export const ERROR = 'boilerplate/Login/ERROR';
export const LOADING = 'boilerplate/Login/LOADING';
export const SUCCESS = 'boilerplate/Login/SUCCESS';
export const LOGIN = 'boilerplate/Login/LOGIN';
export const RESET = 'boilerplate/Login/RESET';
export const GOOGLE_LOGIN = 'boilerplate/Login/GOOGLELOGIN';
export const FACEBOOK_LOGIN = 'boilerplate/Login/FACEBOOKLOGIN';
