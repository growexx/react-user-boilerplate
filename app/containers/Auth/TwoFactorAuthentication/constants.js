/*
 *
 * TwoFactorAuthentication constants
 *
 */

export const CHANGE_VALUE = 'app/TwoFactorAuthentication/CHANGE_VALUE';
export const SUBMIT = 'app/TwoFactorAuthentication/SUBMIT';
export const FORM_KEY = 'twoFactorAuthentication';
export const TEST_OTP_VALUE = 123456;
export const TEST_OTP_VALUE_WITHOUT_LENGTH = 12345;
export const OTP_LENGTH = 6;
