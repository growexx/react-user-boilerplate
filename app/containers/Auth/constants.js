export const AUTH_TYPE = ['LOGIN', 'REGISTER'];

// API-ROUTES
export const API_URL = 'https://b0257f3aa104.ngrok.io';
export const AUTH = 'auth';
export const API_ENDPOINTS = {
  LOGIN: `${API_URL}/${AUTH}/signin`,
  LIST: 'https://randomuser.me/api/?results=5&inc=name,gender,email,nat&noinfo',
  LIST_AVATAR:
    'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
  EXPORT_CSV: '',
};
