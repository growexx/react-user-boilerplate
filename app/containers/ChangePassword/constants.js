/*
 *
 * ChangePassword constants
 *
 */
const PASSWORD_MUST_MATCH = 'Passwords do not match';
export const passwordsMustMatch = (value, allValues) =>
  value !== allValues.newPassword ? PASSWORD_MUST_MATCH : undefined;

export const CHANGE_CURRENT_PASSWORD =
  'boilerplate/ChangePassword/CHANGE_CURRENT_PASSWORD';
export const CHANGE_NEW_PASSWORD =
  'boilerplate/ChangePassword/CHANGE_NEW_PASSWORD';
export const CHANGE_CONFIRM_NEW_PASSWORD =
  'boilerplate/ChangePassword/CHANGE_CONFIRM_NEW_PASSWORD';
export const LOADING = 'boilerplate/ChangePassword/LOADING';
export const SUBMIT_DATA = 'boilerplate/ChangePassword/SUBMIT_DATA';
export const FORM_KEY = 'ChangePassword';
