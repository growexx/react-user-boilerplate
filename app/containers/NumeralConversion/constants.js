export const CREATE_COLUMNS = [
  {
    title: 'Input',
    dataIndex: 'input',
    key: 'input',
  },
  {
    title: 'Value',
    dataIndex: 'value',
    key: 'value',
  },
];
export const FORMAT_COLUMNS = [
  {
    title: 'Number',
    dataIndex: 'number',
    key: 'number',
  },
  {
    title: 'Format',
    dataIndex: 'format',
    key: 'format',
  },
  {
    title: 'String',
    dataIndex: 'string',
    key: 'string',
  },
];

export const NUMBERS_FORMAT = [
  '0,0.0000',
  '0,0',
  '+0,0',
  '0,0.0',
  '0.000',
  '00000',
  '000000,0',
  '0o',
  '0o',
  '0a',
  '0 a',
  '0.0a',
  '0.0[0000]',
  '0.00000',
  '(.00)',
  '.00',
  '(0,0.0000)',
  '0[.]00000',
  '000.00',
];

export const CURRENCY_FORMAT = [
  '$0,0.00',
  '0,0[.]00 $',
  '$ 0,0[.]00',
  '($0,0)',
  '$0.00',
  '($ 0.00 a)',
];

export const BYTES_FORMAT = [
  '0b',
  '0b',
  '0 ib',
  '0.0 b',
  '0.00b',
  '0.00b',
  '0.000 ib',
];

export const PERCENTAGES_FORMAT = ['0%', '0.000%', '0 %', '(0.000 %)'];

export const TIME_FORMAT = ['00:00:00', '00:00:00', '00:00:00'];

export const EXPONENTIAL_FORMAT = ['0,0e+0', '0.00e+0', '0.000e+0'];
