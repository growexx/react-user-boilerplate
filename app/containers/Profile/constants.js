/*
 *
 * Profile constants
 *
 */
export const PROFILE_PLACEHOLDER =
  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum";

export const DATA_TEST_IDS = {
  ABOUT_EDIT: 'About_Edit',
  ABOUT_SAVE: 'About_Save',
  ABOUT_EDITOR: 'About_Editor',
  EDUCATION_EDIT: 'Education_Edit',
  EDUCATION_SAVE: 'Education_Save',
  EXPERIENCE_EDIT: 'Experience_Edit',
  EXPERIENCE_SAVE: 'Experience_Save',
  LICENSEANDCERTIFICATION_EDIT: 'LicensesAndCertifications_edit',
  LICENSEANDCERTIFICATION_SAVE: 'LicensesAndCertifications_save',
};
