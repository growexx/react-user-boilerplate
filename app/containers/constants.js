// Control Bit
export const IS_DEMO_CODE = true;

// App-Routes
export const ROUTES = {
  HOME: '/',
  GITHUB_SEARCH: '/github-search',
  FONT_AWESOME: '/font-awesome',
  LOGIN: '/login',
  REGISTER: '/register',
  PROFILE: '/profile',
  LOGOUT: '/logout',
  LOADER: '/loader',
  EXPORT_DATA: '/export-data',
  USERS: '/users',
  CHARTS: '/charts',
  TEST_ADMIN_PAGE: '/admin',
  UNAUTHORIZED: '/403',
  NUMERAL_CONVERTER: '/number-converter-demo',
  SAMPLE_FORM: '/sample-form',
  CHANGE_PASSWORD: '/change-password',
  TWO_FACTOR_AUTHENTICATION: '/two-factor-authentication',
  FORGOT_PASSWORD: '/forgot-password',
  MULTI_TAB_SUPPORT: '/multi-tab-support',
  PRODUCTS: '/products',
};

// API-ROUTES
export const { API_URL } = process.env;
export const AUTH = 'auth';
export const API_ENDPOINTS = {
  LOGIN: `${API_URL}/${AUTH}/signin`,
  LIST: 'https://randomuser.me/api/?results=5&inc=name,gender,email,nat&noinfo',
  LIST_AVATAR:
    'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
  EXPORT_CSV: '',
  IMAGE_UPLOAD: 'https://www.mocky.io/v2/5cc8019d300000980a055e76',
  FORGOT_PASSWORD: `${API_URL}/${AUTH}/forgetpassword`,
  USERS: `${API_URL}/users`,
};

// Table Pagination default
export const DEFAULT_PAGE = 1;
export const DEFAULT_LIMIT = 10;

/**
 * Roles for system
 * ACL from frontend
 * This configuration is being used in RoleMiddleware
 */
// Note: Define role in this constant
export const ROLES = {
  USER: 1,
  ADMIN: 10,
};

// Note: Add Routes which needs to check role before it's access
export const RESTRICTED_ROUTES = [ROUTES.PROFILE, ROUTES.TEST_ADMIN_PAGE];

// Note: Define role against all accessible routes
export const ROLE_BASED_ROUTE_ACCESS = {
  [ROLES.USER]: [ROUTES.PROFILE],
  [ROLES.ADMIN]: [ROUTES.TEST_ADMIN_PAGE],
};

// Note: Role based default routes for redirection
export const ROLE_BASED_DEFAULT_ROUTE = {
  [ROLES.USER]: ROUTES.HOME,
};

/**
 * Controls which sidebar entries shown to this user
 */
// Sidebar role based update
export const ROLE_BASED_SIDEBAR_MENU = {
  [ROLES.USER]: [],
  [ROLES.ADMIN]: [ROUTES.TEST_ADMIN_PAGE],
};

export const DEFAULT_PAGE_NO = 1;
export const DEFAULT_SIZE = 10;
export const SORTING = {
  ASC: 1,
  DESC: -1,
  ascend: 1,
  descend: -1,
};
/**
 *
 * @param {string} order  ['ascend','descend']
 * @returns
 */
export const GET_SORT_ORDER = order => {
  switch (order) {
    case 'ascend':
      return SORTING.ASC;
    case 'descend':
      return SORTING.DESC;
    default:
      return SORTING.DESC;
  }
};

export const GET_DEFAULT_PAGINATION = () => ({
  pageSize: DEFAULT_SIZE,
  current: DEFAULT_PAGE_NO,
  total: 0,
});

export const GENERIC_MOMENT_DATE_FORMAT = 'YYYY-MM-DD';
export const FULL_GENERIC_MOMENT_DATE_FORMAT = 'YYYY-MM-DD hh:mm:ss a';
export const CHANNEL_NAME = 'DATA_CHANNEL_NAME';
