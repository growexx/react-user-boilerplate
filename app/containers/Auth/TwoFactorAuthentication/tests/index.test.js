/**
 *
 * Tests for TwoFactorAuthentication
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
import history from 'utils/history';
import { ConnectedRouter } from 'connected-react-router';
import { TwoFactorAuthentication, mapDispatchToProps } from '../index';
import Lodable from '../Loadable';
import { TEST_OTP_VALUE, TEST_OTP_VALUE_WITHOUT_LENGTH } from '../constants';
import configureStore from '../../../../configureStore';
let store;
const componentWrapper = Component =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Component />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );
describe('<TwoFactorAuthentication />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper(TwoFactorAuthentication);
    expect(firstChild).toMatchSnapshot();
  });
  it('mapDispatch to props', () => {
    const mockFn = jest.fn();
    const returnValue = mapDispatchToProps(mockFn);
    // when otp length is achieved
    returnValue.onChangeValue(TEST_OTP_VALUE.toString());
    // when otp length is not achieved
    returnValue.onChangeValue(TEST_OTP_VALUE_WITHOUT_LENGTH.toString());
    expect(mockFn).toBeCalled();
  });
  it('Should render and match the snapshot Loadable', () => {
    const {
      container: { firstChild },
    } = componentWrapper(Lodable);
    expect(firstChild).toMatchSnapshot();
  });
});
