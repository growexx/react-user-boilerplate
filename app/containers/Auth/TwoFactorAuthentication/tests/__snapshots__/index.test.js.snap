// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwoFactorAuthentication /> Should render and match the snapshot 1`] = `
.c2 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .otpComponentContainerStyle {
  width: 100%;
}

.c2 .otpComponentInputStyle {
  color: #4d186e;
  border: 1px solid #d9d9d9;
  width: 100% !important;
  height: 40px;
  border-radius: 10px;
}

.c2 .otpComponentFocusStyle {
  border: 1px solid #4d186e;
  outline: none;
  border-radius: 10px;
}

.c2 .otpComponentDisabledStyle {
  background: #f5f5f5;
}

.c2 .otpComponentErrorStyle {
  border: 1px solid #ff0000;
}

.c1 {
  font-family: 'Montserrat',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
}

.c1 .twoFactorAuthenticationTitle {
  text-align: center;
  margin: 0;
  font-weight: 700;
  font-size: 42px;
  color: #4d186e;
}

.c0 {
  font-family: 'Montserrat',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 .sideContainer {
  max-width: 513px;
  width: 100%;
  background: #4d186e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 32px;
}

.c0 .sideContainer .registrationSideContainer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.c0 .sideContainer .registrationSideContainer .title {
  text-align: center;
  font-size: 42px;
  font-weight: 700;
  margin: 0;
}

.c0 .sideContainer .registrationSideContainer .subTitle {
  width: 60%;
  text-align: center;
  margin: 0;
}

.c0 .sideContainer .registrationSideContainer button {
  margin-top: 35px;
  height: 40px;
  width: 160px;
  background: transparent;
  border: 2px solid #fff;
  border-radius: 50px;
  text-align: center;
  color: #fff;
}

.c0 .sideContainer .registrationSideContainer button:hover {
  color: #fff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="sideContainer"
    >
      <div
        class="loginContainer"
      >
        <div
          class="registrationSideContainer"
        >
          <p
            class="title"
          >
            <span>
              New Here!
            </span>
          </p>
          <p
            class="subTitle"
          >
            <span>
              To join us, please sign up with your personal info
            </span>
          </p>
          <a
            href="/register"
          >
            <button
              class="ant-btn"
              type="button"
            >
              <span>
                SIGN UP
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <p
        class="twoFactorAuthenticationTitle"
      >
        <span>
          Two Factor Authentication
        </span>
      </p>
      <div
        class="c2"
      >
        <div
          class="otpComponentContainerStyle"
          style="display: flex;"
        >
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Please enter verification code. Digit 1"
              autocomplete="off"
              class="otpComponentInputStyle otpComponentFocusStyle"
              maxlength="1"
              placeholder="0"
              style="width: 1em; text-align: center;"
              type="tel"
              value=""
            />
            <span>
              -
            </span>
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Digit 2"
              autocomplete="off"
              class="otpComponentInputStyle"
              maxlength="1"
              placeholder="0"
              style="width: 1em; text-align: center;"
              type="tel"
              value=""
            />
            <span>
              -
            </span>
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Digit 3"
              autocomplete="off"
              class="otpComponentInputStyle"
              maxlength="1"
              placeholder="0"
              style="width: 1em; text-align: center;"
              type="tel"
              value=""
            />
            <span>
              -
            </span>
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Digit 4"
              autocomplete="off"
              class="otpComponentInputStyle"
              maxlength="1"
              placeholder="0"
              style="width: 1em; text-align: center;"
              type="tel"
              value=""
            />
            <span>
              -
            </span>
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Digit 5"
              autocomplete="off"
              class="otpComponentInputStyle"
              maxlength="1"
              placeholder="0"
              style="width: 1em; text-align: center;"
              type="tel"
              value=""
            />
            <span>
              -
            </span>
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Digit 6"
              autocomplete="off"
              class="otpComponentInputStyle"
              maxlength="1"
              placeholder="0"
              style="width: 1em; text-align: center;"
              type="tel"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<TwoFactorAuthentication /> Should render and match the snapshot Loadable 1`] = `null`;
